### ISSUE TITLE
Remove requirement on RegEx

### ISSUE THREAD

**Issue by TomFaulkner** [TomFaulkner/SenseMe#5]
The regex call, which appears in both discovery functions, is unnecessary, switch to split on ;

**Comment by shepherdjay** [comment]
I would be willing to take this up, do you have an example string you are currently running the regex against so I can write a test?

**Comment by TomFaulkner** [comment]
That's a good question. I can see where the three are, but as for the string, I don't think I have an example of that at the moment. Do you have a device to run it on? If so, you could run a few commands and log the output (might have to add logging for that) to see them.

Tests are good, I've been meaning to add unit tests for some time!

If you don't have a device, I can get back to you in a couple of days with some strings.

**Comment by shepherdjay** [comment]
I don't have a device myself, but can get started on the code changes I would intend to make and just insert whatever test strings you get back to me with into the test. Since you don't have a test framework yet for this project am I good implementing in pytest or would you prefer the built in unittest. I can do it in either.

**Comment by TomFaulkner** [comment]
I have been meaning to try pytest. Briefly looking at the docs for it I think that will be fine.

There isn't a linter config, but I am using black's autoformatting. (https://github.com/ambv/black)

**Comment by shepherdjay** [comment]
Apologize for the misunderstanding, do you have an example of the raw messages that are received on the buffer? The regex currently does a match for messages that include DEVICE;ID which makes me wonder if other types of messages are received.

So this line here:
`message = s.recvfrom(1024)`

Maybe modify to add something that logs that message like this:

```
LOGGER.info("Received a message")
LOGGER.debug(f"Message: {message}")
```

**Comment by TomFaulkner** [comment]
Looking back at the code, you're right in that the examples I gave above aren't sufficient. The ones I gave are status updates, which don't have the same content as some of the other responses. The lines above are output from what is now line 1293, the `_query` method, on the master branch (which was updated last night).

The lines I think you are looking for is the discover and discover_single (which are sadly duplicate of one another.) I'll get back to you with output from those.

**Comment by shepherdjay** [comment]
Yeah looking to just get what the socket sees. So that `s.recvfrom(1024)` in discover being assigned to `message` is what will ultimately be passed to the new function to be decoded and parsed. I did see the master update which I'll have to pull in and fix conflicts before I make another push on the pull request. Depending on what those messages look like regex may still be the best way to parse it but this will at least refactor out the duplicate code.

**Comment by shepherdjay** [comment]
Here is a working refactored function example. Not sure if this is more readable than regex, let me know your thoughts:

```
def decode_discovery(message: Tuple[bytes, Tuple[str, str]]) -> Tuple:
    """ Takes a message in the return format of socket.recvfrom and returns decoded SenseMe discovery information """
    message_decoded = message[0].decode("utf-8")
    message_items = message_decoded.strip('()').split(';')
    name = message_items[0]
    mac = message_items[3]
    model, series = message_items[4].split(',')
    ip = message[1][0]
    return name, mac, model, series, ip
```

**Comment by TomFaulkner** [comment]
Sorry to not get back to you, as I said in the PR comments.

The code looks good, I'll have to compare it to a string from the fan. It will probably be next week when I can get them to you.
